pkg/tests/tasks/security/certmanager/main_test.go

```diff
@@ -71,10 +76,20 @@ func setupCertManagerOperator(t test.TestHelper) {
 	oc.CreateNamespace(t, certManagerOperatorNs)
 
 	t.LogStep("Install cert-manager-operator")
-	oc.ApplyString(t, certManagerOperatorNs, certManagerOperator)
-	oc.WaitPodReady(t, pod.MatchingSelector("name=cert-manager-operator", certManagerOperatorNs))
-	oc.WaitPodReady(t, pod.MatchingSelector("app=cert-manager", certManagerNs))
+	oc.ApplyTemplate(t, certManagerOperatorNs, certManagerOperator, map[string]string{"Version": certmanagerVersion})
+	waitOperatorSucceded(t, certManagerOperatorNs)
 
 	t.LogStep("Create root ca")
 	oc.ApplyString(t, certManagerNs, rootCA)
 }
+
+func waitOperatorSucceded(t test.TestHelper, certManagerOperatorNs string) {
+	t.Log("Waiting for cert-manager-operator to succeed")
+	retry.UntilSuccess(t, func(t test.TestHelper) {
+		shell.Execute(t,
+			fmt.Sprintf("oc get csv %s -n %s -o jsonpath='{.status.phase}'", certmanagerVersion, certManagerOperatorNs),
+			assert.OutputContains("Succeeded", "Operator ready", "Failed to wait for cert-manager-operator to succeed"))
```

Comment:
You can use `oc wait --for=jsonpath='{.status.phase}'=Succeeded` instead. And this means you can use `oc.Wait()` instead of `shell.Execute`.

Reply:
Ok, we do not have a oc.Wait general but I change the Wait Condition for WaitFor to extend the use and be able to set more than only --for condition=xxxxx in this case the func now will be able to accept the example that you give me. I did not create a new separated func because the code is very similar (To avoid repeat the code)

Reply:
Actually, making a change that requires you to modify a bunch of files like this is not the best approach. It would be, if the change was to simplify those files, but you've actually made the usage more complex. Now people need to know the syntax they need to use (e.g. "condition=Ready"), whereas previously, we just had "Ready" or even better - we have `oc.WaitReady` (at least I think we have). 

So, it would be better to create `oc.WaitForPhase` (or `oc.WaitPhase`) instead of changing the existing `oc.WaitCondition`. Of course both of these functions could then internally call `oc.WaitFor()`, so there would be no duplicated code.

Reply:
Ok, understood. We need to merge this because we need to start run the test for the release, I'll make the changes and submit a PR and track all the work in a new Jira ticket: https://issues.redhat.com/browse/OSSM-4309